### PR TITLE
fix(core/platform): time-picker - a11y issues

### PR DESCRIPTION
**Pull request by dmitry-stepanenko**
## Related Issue(s)

closes https://github.com/SAP/fundamental-ngx/issues/6142
closes https://github.com/SAP/fundamental-ngx/issues/4669

## Description
addressing a11y issues for time picker component in core &. platform

#### Overview of the changes made as per each of 2 tickets
4669:
- issue 1 - cannot reproduce
- issue 2 - done
- issue 3 - will be addressed apart in fundamental-styles https://github.com/SAP/fundamental-styles/pull/3416
- issue 4 - not done, the color is according to specs
- issue 5 - cannot reproduce
- issue 6 - done
- issue 7 - cannot reproduce
- issue 8 - done
- issue 9 - done
- issue 10 - not an issue, a11y software simply points to the fact input text is selected
- issue 11 - cannot reproduce
- issue 12 - cannot reproduce
- issue 12 (2) - cannot reproduce
- issue 13 - cannot reproduce
- issue 13 (2) - done
- issue 14 - done
6142:
- issue 1 - cannot reproduce
- issue 2 - cannot reproduce
- issue 3 - cannot reproduce

##### During Implementation

1. Visual Testing:

-   [n/a] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [n/a] RTL/LTR - proper rendering and labeling
-   [n/a] responsiveness(resize)
-   [n/a] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [n/a] API boundary values
-   [n/a] different combinations of components - free style
-   [n/a] change the API values during testing

3. Documentation and Example validations

-   [n/a] missing API documentation or it is not understandable
-   [n/a] poor examples
-   [n/a] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [n/a] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [n/a] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

**Review by InnaAtanasova — APPROVED** [review]

**Comment by Karthickc1994** [comment]
Hello Colleague,

Retest has been taken below are the results.

closes https://github.com/SAP/fundamental-ngx/issues/6142 --> closed
closes https://github.com/SAP/fundamental-ngx/issues/4669 --> Reopened.

Regards,
Karthikeyan.c

File: libs/platform/src/lib/form/form-group/form-field/form-field.component.ts [hunk 1]
```diff
@@ -94,7 +96,7 @@ export class FormFieldComponent implements FormField, AfterContentInit, AfterVie
      * Form field id
      */
     @Input()
-    id: string;
+    id: string = (++defaultId).toString();
```
Comment (platon-rov) [inline libs/platform/src/lib/form/form-group/form-field/form-field.component.ts:99]: having id=1 (example) is very fragile on my opinion, could we extend it?

Comment (platon-rov) [reply libs/platform/src/lib/form/form-group/form-field/form-field.component.ts:99]: oh, i see, it gets extended in the template. but value of this field used as a suffix, could we mention it in the typedoc?

Comment (dmitry-stepanenko) [reply libs/platform/src/lib/form/form-group/form-field/form-field.component.ts:99]: actually it felt more reasonable to use the whole id than a suffix, updated

**Comment by dmitry-stepanenko** [comment]
> Hello Colleague,
> 
> Retest has been taken below are the results.
> 
> closes #6142 --> closed closes #4669 --> Reopened.
> 
> Regards, Karthikeyan.c

@Karthickc1994 fixed, please review

**Comment by dmitry-stepanenko** [comment]
Verified as per https://github.com/SAP/fundamental-ngx/issues/4669#issuecomment-1168350088

**Pull request merged**